### PR TITLE
Add Logic for Raw XML

### PR DESCRIPTION
**Pull request by kjdowney**
Resolve issues #10 and #16 by adding logic to capture the raw XML of a section that doesn't otherwise
have handling, instead of throwing a NotImplementedError.

### CODE REVIEW

**Comment by malcolm-brooks** [comment]
It is possible to get this merged? It would be a really useful addition!

**Comment by ktdreyer** [comment]
Thanks Kyle! Please accept my apologies for taking so long to merge this PR. I've pushed it to PyPI as v1.0.7.

**Comment by selurvedu** [comment]
Woah, this is superb! This tool is now much more easy to use! Thanks a lot!

**Comment by kjdowney** [comment]
@ktdreyer Glad it could be useful!

I thought I had a little write-up that would explain the philosophy behind the XML output in the original commit message, but I guess I didn't. So, to elaborate a little on the changes and the reasoning, I offer the following:

## The Motivation
Like most folks that I imagine find Jenkins Job Wrecker, I stumbled across the utility when looking for a way to start housing all the Jenkins jobs I was working with under source control. Jenkins Job Builder was an attractive solution, but most of the jobs had been created with through the UI and the thought of rewriting *everything* was a bit daunting. Additionally, there were some Jenkins plugins utilized on our instance that aren't heavily utilized by others, which I knew would cause some issues with mapping.

While not a perfect solution, Jenkins Job Builder having the ability to ingest raw XML was considered a good enough trade-off so that we wouldn't have to spend time upfront refactoring every job into a format we could easily expand and/or refactor later.

## The Problem
Overall, Jenkins Job Wrecker does a great job of generating Jenkins Job Builder YAML files, with the exception of handling plugins that it doesn't know about. As pointed out in #10 and #16, Jenkins Job Wrecker didn't really have a graceful way of handling things that it didn't know about, other than throwing a `NotImplementedError` exception and stopping processing. While this was typically manifested by JJW trying to process a plugin that it didn't have logic to map, I also ran into an issue where JJB didn't have a mapping for an attribute, thereby making it impossible for JJW to perform any valid translation at all.

My desire was for something that could take years worth of work and archive it in a format that's friendlier to being modified, accepting the possibility that the YAML output might be tainted in places with raw XML.

## The Solution
The logic that was implemented tries to stay out of the way by wrapping each of the different translation logic sections in a `try` statement with an `except` block that handles the case of a `NotImplementedError` that is expected to be thrown by the translation logic functions when encountering something they don't know about. If the `NotImplementedError` exception is caught, the *entire* XML element is inserted for output as raw XML. This approach is also taken even if a partial translation of the XML element is possible to err on the side of caution. The idea was that if the element/attribute was important enough to be on the source system, we should keep it going forward with the raw XML in the source files indicating areas that might need a little extra attention moving forward.

While a little coarse in granularity, I was able to convert a large number of jobs with (what I feel is) a very small amount of change to the original structure.

## The Followup
A new question that might be raised is whether this should always be enabled. At the time of development, my main concern was getting a large number of jobs converted and into source control before worried about how much (if at all) they would need to be refactored. The focus was on getting output, rather than making sure that the output had been fully translated. What I perhaps failed to take into consideration at the time was that these changes were modifying the default (and only) behavior. I could see how someone could argue a need for making sure that the output is 100% translated and wanting an error (or warning) in the case that a translation does not exist and would result in raw XML being injected into the definition file.

## TL;DR
Modified default behavior to emphasize getting an output, even if it means raw XML in the definition file; May require an option to enable old behavior if this causes issues.

**Comment by malcolm-brooks** [comment]
This should resolve #16 and resolve #10.
